pkg/snapshot/restorer/restorer.go

```diff
@@ -796,31 +796,32 @@ func verifySnapshotRevision(clientKV client.KVCloser, snap *brtypes.Snapshot) er
 	return nil
 }
 
-// getDecompressedSnapshotReadCloser passes the given ReadCloser through the
+// getNormalizedSnapshotReadCloser passes the given ReadCloser through the
 // snapshot decompressor if the snapshot is compressed using a compression policy.
-// Also returns whether the snapshot was initially compressed or not, as well as
+// If snapshot is not compressed, it returns the given ReadCloser as is.
+// It also returns whether the snapshot was initially compressed or not, as well as
 // the compression policy used for compressing the snapshot.
-func getDecompressedSnapshotReadCloser(rc io.ReadCloser, snap *brtypes.Snapshot) (io.ReadCloser, bool, string, error) {
+func getNormalizedSnapshotReadCloser(rc io.ReadCloser, snap *brtypes.Snapshot) (io.ReadCloser, bool, string, error) {
 	isCompressed, compressionPolicy, err := compressor.IsSnapshotCompressed(snap.CompressionSuffix)
 	if err != nil {
-		return nil, false, compressionPolicy, err
+		return rc, false, "", err
 	}
 
 	if isCompressed {
 		// decompress the snapshot
 		rc, err = compressor.DecompressSnapshot(rc, compressionPolicy)
 		if err != nil {
-			return nil, true, compressionPolicy, fmt.Errorf("unable to decompress the snapshot: %v", err)
+			return rc, true, compressionPolicy, fmt.Errorf("unable to decompress the snapshot: %v", err)
 		}
 	}
 
-	return rc, true, compressionPolicy, nil
+	return rc, false, "", nil
```

Comment:
This returns `false` even if `isCompressed` is true and `compressor.DecompressSnapshot` succeeded.

It can just return `isCompressed`

Reply:
Ah yes, you're right. I forgot to handle the case where decompression actually succeeds 😂. I've made the change now, thanks!